docs/torch/node_class.rst

```diff
@@ -34,261 +44,221 @@ DeepGNN models typically contain multiple parts:
-	...         labels = torch.squeeze(labels)              # [N]
-	...         edges = torch.squeeze(edges)                # [X, 2], X: num of edges in subgraph
-	...         edges_value = torch.squeeze(edges_value)    # [X]
-	...         adj_shape = torch.squeeze(adj_shape)        # [2]
-	...
-	...         sp_adj = torch.sparse_coo_tensor(edges, edges_value, adj_shape.tolist())
-	...         h_1 = self.input_layer(feat, sp_adj)
-	...         scores = self.out_layer(h_1, sp_adj)
-	...
-	...         labels = labels.type(torch.int64)
-	...         labels = labels[mask]  # [batch_size]
-	...         scores = scores[mask]  # [batch_size]
-	...         pred = scores.argmax(dim=1)
-	...         loss = self.xent(scores, labels)
-	...         return loss, pred, labels
-
-Model Init
-==========
-We need to implement `create_model` and `create_optimizer` functions to allow distributed workers initialize model and optimizer.
+    >>> class GAT(BaseModel):
+    ...     def __init__(
+    ...         self,
+    ...         in_dim: int,
+    ...         head_num: List = [8, 1],
+    ...         hidden_dim: int = 8,
+    ...         num_classes: int = -1,
+    ...         ffd_drop: float = 0.0,
+    ...         attn_drop: float = 0.0,
+    ...     ):
+    ...         super().__init__(np.float32, 0, 0, None)
+    ...         self.num_classes = num_classes
+    ...         self.out_dim = num_classes
+    ...
+    ...         self.input_layer = GATConv(
+    ...             in_dim=in_dim,
+    ...             attn_heads=head_num[0],
+    ...             out_dim=hidden_dim,
+    ...             act=F.elu,
+    ...             in_drop=ffd_drop,
+    ...             coef_drop=attn_drop,
+    ...             attn_aggregate="concat",
+    ...         )
+    ...         layer0_output_dim = head_num[0] * hidden_dim
+    ...         assert len(head_num) == 2
+    ...         self.out_layer = GATConv(
+    ...             in_dim=layer0_output_dim,
+    ...             attn_heads=head_num[1],
+    ...             out_dim=self.out_dim,
+    ...             act=None,
+    ...             in_drop=ffd_drop,
+    ...             coef_drop=attn_drop,
+    ...             attn_aggregate="average",
+    ...         )
+    ...
+    ...     def forward(self, context: Dict[Any, np.ndarray]):
+    ...         nodes = torch.squeeze(context["nodes"])                # [N], N: num of nodes in subgraph
+    ...         feat = torch.squeeze(context["feat"])                  # [N, F]
+    ...         mask = torch.squeeze(context["input_mask"])            # [N]
+    ...         labels = torch.squeeze(context["labels"])              # [N]
+    ...         edges = torch.squeeze(context["edges"].reshape((-1, 2)))                # [X, 2], X: num of edges in subgraph
+    ...
+    ...         edges = np.transpose(edges)
+    ...
+    ...         # TODO This is not stable, when doing batch_size < graph size ends up with size < index values. use torch.unique to remap edges
```

Comment:
todo in this PR?

Reply:
It's an issue with GAT model not this example specifically, removed.